### PR TITLE
ramips:Add support for Thunder Timecloud

### PR DESCRIPTION
**Pull request by 981213**
Thunder Timecloud is a small NAS with MT7621A.
It has 1 USB port and an SD Card slot.There is no wireless cards.

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

### CODE REVIEW

**Comment by diizzyy** [comment]
Do you have a link to the product page and doesn't the SD-driver crash/panic on your box?

**Comment by 981213** [comment]
Ah...There is only a product page in Simplified Chinese: http://xzb.xunlei.com/

In fact I haven't tested the SD card yet.:(
I forgot to add the driver in my test firmware and I just add the default packages before sending this pull request.

**Comment by ignisf** [comment]
Specs here: http://xzb.xunlei.com/#/param

**Comment by 981213** [comment]
I've just added the SD card module and it crashed:(
Ah...Do I need to remove the SD card module from the default packages and send another pull request?
`
[   64.340000] INFO: rcu_sched self-detected stall on CPU
[   64.340000] 	0-...: (1 GPs behind) idle=625/140000000000002/0 softirq=123/124 fqs=5995 
[   64.340000] 	 (t=6000 jiffies g=-248 c=-249 q=775)
[   64.340000] Task dump for CPU 0:
[   64.340000] kmodloader      R running      0   319      1 0x08100004
[   64.340000] Stack : 00000000 80063a1c 00000000 0000000e 00000000 80063a1c 00000000 00000006
	  00000006 800622d8 00000000 00000000 00000000 00000000 80450000 804c0000
	  00000000 8f9567e8 000000e0 80450000 00000000 00000000 00000000 00000000
	  00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
	  00000000 00000000 00000000 00000000 00000000 dc8ba60f 00000000 8f9567e8
	  ...
[   64.340000] Call Trace:
[   64.340000] [<8001665c>] show_stack+0x50/0x84
[   64.340000] [<8006d834>] rcu_dump_cpu_stacks+0x98/0xe8
[   64.340000] [<80070fa4>] rcu_check_callbacks+0x29c/0x720
[   64.340000] [<80072eec>] update_process_times+0x34/0x70
[   64.340000] [<8008211c>] tick_sched_timer+0x238/0x2a0
[   64.340000] [<80073f40>] __hrtimer_run_queues+0x10c/0x1d4
[   64.340000] [<80074198>] hrtimer_interrupt+0xec/0x2ac
[   64.340000] [<8026c868>] gic_compare_interrupt+0x2c/0x40
[   64.340000] [<800691f8>] handle_percpu_devid_irq+0xac/0xf4
[   64.340000] [<80064908>] generic_handle_irq+0x24/0x3c
[   64.340000] [<801d83f0>] gic_handle_local_int+0x98/0xec
[   64.340000] [<801d85cc>] gic_irq_dispatch+0x10/0x20
[   64.340000] [<80064908>] generic_handle_irq+0x24/0x3c
[   64.340000] [<8001349c>] do_IRQ+0x1c/0x2c
[   64.340000] [<801d7918>] plat_irq_dispatch+0xb4/0xdc
[   64.340000] [<80005430>] ret_from_irq+0x0/0x4
[   64.340000] [<8002e8a0>] __do_softirq+0xc0/0x2e0
[   64.340000] [<8002ed3c>] irq_exit+0x68/0x84
[   64.340000] [<801d7918>] plat_irq_dispatch+0xb4/0xdc
[   64.340000] [<80005430>] ret_from_irq+0x0/0x4
[   64.340000] [<8000ef40>] _raw_spin_unlock_irqrestore+0x24/0x2c
[   64.340000] [<800673f4>] __setup_irq+0x4e4/0x678
[   64.340000] [<80067748>] request_threaded_irq+0x11c/0x18c
[   64.340000] [<8f3a2df4>] init_module+0x6edf4/0x6ef40 [mtk_sd]
[   64.340000] 
[   79.500000] INFO: rcu_sched detected expedited stalls on CPUs/tasks: { 0-... } 6000 jiffies s: 1
[   79.500000] Task dump for CPU 0:
[   79.510000] kmodloader      R running      0   319      1 0x08100004
[   79.510000] Stack : 00000000 00000000 00000000 8120e7e0 8f9567e8 00000162 00000800 8120e7e0
	  02da6173 8004fde4 0000013f 00000001 08100004 804987e0 8f9567e8 80450880
	  00000000 800549ec 803d7ddc 8f2f36cc 8120e7e0 80061938 ffffffff 00000000
	  8f956830 00000000 00000000 00000000 8120e818 00000000 00000000 00000000
	  000003f8 00000000 000007e7 00000000 000007e7 00000000 8120e7e0 00000000
	  ...
[   79.550000] Call Trace:
[   79.550000] [<8000bc90>] __schedule+0x574/0x758
[   79.560000] 
`

**Comment by neheb** [comment]
You can remove the sdhci kernel driver from the defaults and just make another commit. Then just "git rebase -i HEAD~2", squash, and "git push --force"

In other news, I tried this on my mt7621 device after editing the DTS and same result. Driver remains broken.

**Comment by 981213** [comment]
OK.I've just removed the sdcard driver from the defaults:)

File: target/linux/ramips/image/mt7621.mk [hunk 1]
```diff
@@ -105,6 +105,13 @@ define Device/sap-g3200u3
 endef
 TARGET_DEVICES += sap-g3200u3
 
+define Device/timecloud
+  DTS := Timecloud
+  DEVICE_TITLE := Thunder Timecloud
+  DEVICE_PACKAGES := kmod-usb3
+endef
```
Comment (blogic) [inline target/linux/ramips/image/mt7621.mk]: i doubt that there is usb3 on this board.

**Comment by blogic** [comment]
merged, thanks !

**Pull request closed without merging**